ocp3/ocp3/Team.swift

```diff
@@ -16,14 +16,74 @@ class Team {
         self.name = name
         self.characters = characters
     }
+    //////////////////////////////////////////Fonction qui donne les info sur les membres de l'équipe
     func teamInfo() {
         print("L'équipe \(self.name) est composée de :")
-        for character in characters{
+        for character in characters {
             print ("- \(character.name) de type \(character.type) avec une vie à : \(character.life)")
         }
     }
+    //////////////////////////////////////////Fonction qui permet de choisir avec quel personnage on attaque (En bouclant si le choix est un personnage mort)
+    func chooseAttacker() -> Character {
+        print("Equipe \(self.name) veuillez choisir un de vos personnages pour passer à l'action (de 1 à 3)")
+        miniInfo()
+        if let choice = Int(readLine()!) {
+            if self.characters[choice - 1].life == 0 {
+                print("Ce personnage est mort.Veuillez en choisir un autre.")
+                return chooseAttacker()
+            } else {
+                switch choice {
+                case 1: return self.characters[0]
+                case 2: return self.characters[1]
+                case 3: return self.characters[2]
+                default: print("Veuillez choisir entre 1 et 3")
+                }
+            }
+        }
+        return chooseAttacker()
+    }
+    ///////////////////////////////////////////Fonction qui permet de choisir quel personnage on va attaquer (En bouclant si le choix est un personnage mort)
+    func chooseTarget() -> Character {
+        print("Veuillez choisir quel personnage attaquer (de 1 à 3)")
+        miniInfo()
+        if let choice = Int(readLine()!) {
```

Comment:
You have simply check if the readline is a Int but not if it's between 1 and 3.
Exemple : you can try with 4 it will crash

Reply:
Done